src/server/rdb_load.cc

```diff
@@ -1892,6 +1894,30 @@ auto RdbLoaderBase::ReadStreams() -> io::Result<OpaqueObj> {
   SET_OR_UNEXPECT(LoadLen(nullptr), load_trace->stream_trace->ms);
   SET_OR_UNEXPECT(LoadLen(nullptr), load_trace->stream_trace->seq);
 
+  if (rdbtype >= RDB_TYPE_STREAM_LISTPACKS_2) {
+    /* Load the first entry ID. */
+    SET_OR_UNEXPECT(LoadLen(nullptr), load_trace->stream_trace->first_id.ms);
+    SET_OR_UNEXPECT(LoadLen(nullptr), load_trace->stream_trace->first_id.seq);
+
+    /* Load the maximal deleted entry ID. */
+    SET_OR_UNEXPECT(LoadLen(nullptr), load_trace->stream_trace->max_deleted_entry_id.ms);
+    SET_OR_UNEXPECT(LoadLen(nullptr), load_trace->stream_trace->max_deleted_entry_id.seq);
+
+    /* Load the offset. */
+    SET_OR_UNEXPECT(LoadLen(nullptr), load_trace->stream_trace->entries_added);
+  } else {
+    /* During migration the offset can be initialized to the stream's
+     * length. At this point, we also don't care about tombstones
+     * because CG offsets will be later initialized as well. */
+    load_trace->stream_trace->max_deleted_entry_id.ms = 0;
+    load_trace->stream_trace->max_deleted_entry_id.seq = 0;
+    load_trace->stream_trace->entries_added = load_trace->stream_trace->stream_len;
+
+    /* Since the rax is already loaded, we can find the first entry's * ID. */
+    // TODO add implementation
```

Comment:
it's not clear from TODO what needs to be implemented. What are the the values we read but discarded. Please expand on it.